plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -243,20 +254,36 @@ private Map<String, ColumnHandle> makeColumnHandles(List<IndexMetadata.Field> fi
         for (BuiltinColumns builtinColumn : BuiltinColumns.values()) {
             result.put(builtinColumn.getName(), builtinColumn.getColumnHandle());
         }
-
         for (IndexMetadata.Field field : fields) {
             TypeAndDecoder converted = toTrino(field);
             result.put(field.name(), new OpenSearchColumnHandle(
-                    field.name(),
+                    ImmutableList.of(field.name()),
                     converted.type(),
                     field.type(),
                     converted.decoderDescriptor(),
-                    supportsPredicates(converted.type)));
+                    supportsPredicates(field.type(), converted.type())));
         }
 
         return result.buildOrThrow();
     }
 
+    private static boolean supportsPredicates(IndexMetadata.Type type, Type trinoType)
+    {
+        if (type instanceof DateTimeType) {
+            return true;
+        }
+        if (type instanceof ObjectType) {
+            return false;
+        }
+
+        return switch (trinoType) {
+            case TimestampType _, BooleanType _, TinyintType _, SmallintType _, IntegerType _, BigintType _, RealType _ -> true;
+            case DoubleType _ -> !(type instanceof ScaledFloatType);
+            case VarcharType _ -> ((PrimitiveType) type).name().toLowerCase(ENGLISH).equals("keyword");
```

Comment:
This will throw if type is not a PrimitiveType. Change it to:

```java
case VarcharType _ when type instanceof PrimitiveType primitiveType && primitiveType.name().toLowerCase(ENGLISH).equals("keyword") -> true;
```

Reply:
Agreed, thanks!